### PR TITLE
Release 0.2.0-rc.1

### PR DESCRIPTION
**Pull request by jpengar**
#### What type of PR is this?

* subproject management

#### What this PR does / why we need it:

This is "Release PR" for ICM Release 0.2.0-rc.1

#### Which issue(s) this PR fixes:

Related to #146

#### Special notes for reviewers:

* **The team agreed to proceed with a release candidate directly, bypassing the alpha release, given the stability and closed scope of the current state. See https://wiki.camaraproject.org/x/JYDcAQ for further details.**
* `README.md` - with the existing format in ICM there is no need to change anything.
* `CAMARA-AuthN-AuthZ-Concept.md` is DEPRECATED and needs to be removed, but we should do it for the public release I think, not now for this release candidate "pre-release" (to avoid "breaking" the main branch now).
* References in the released documents should always be links to a released version, not to the main branch. If the documents are released together, the links need to be updated at least within the release PR. But, same as previous point, we should do it for the public release, I think, and not now for this release candidate "pre-release" (to avoid "breaking" the main branch now).
* `CHANGELOG.md` - Only the CHANGELOG file needs to be updated in the pre-release for now. Added all changes from the previous public release v0.1.0. 

NEXT STEPS (edited by @hdamker):

- Merge this PR when ready.
- Merge #173 as agreed in [the June 19 WG meeting call](https://wiki.camaraproject.org/pages/viewpage.action?pageId=31227941). **→ UPDATE (21/06): DONE**
~~- Create release v0.2.0 branch: `release-0.2.0-rc`~~
- Create a release `v0.2.0-rc.1` tag for the latest commit.
- Create pre-release v0.2.0-rc.1 in the github project (copy the section from CHANGELOG.md into the description)

NOTE: I will be on vacation for the next two weeks, I will need @AxelNennker, @sebdewet or other volunteers to continue this work for me. My goal is to merge this PR before I leave :) (fingercross)

#### Changelog input

```
ICM Release 0.2.0-rc.1
```

#### Additional documentation 

https://wiki.camaraproject.org/display/CAM/Meta-release+Process#MetareleaseProcess-Milestoneactivities
https://wiki.camaraproject.org/display/CAM/Meta-release+Fall24

### CODE REVIEW

**Review by diegogonmar — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,51 @@
 # Changelog Identity And Consent Management
 
+## Table of Contents
+
+- [v0.2.0-rc](#v020-rc)
+- [v0.1.0 - Initial version](#v010---initial-version)
+
+# v0.2.0-rc
```
Comment (rartych) [inline CHANGELOG.md:8]: According to  https://wiki.camaraproject.org/display/CAM/Meta-release+Process#MetareleaseProcess-Versionandreleasenumbering the version number should follow the template `x.y.z-rc.n`

```suggestion
# v0.2.0-rc.1
```
Then the links and subsequent occurrences should be modified accordingly.

Comment (diegogonmar) [reply CHANGELOG.md:8]: @sebdewet @AxelNennker, Jesús is on holiday and as I am not codeowner I cannot send a commit. Can you do it for me?, I'm fine with this correction. The change is a replace all _0-rc_ to _0-rc.1_, except in line 5 where has to be _0-rc1_ to make link work. Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,51 @@
 # Changelog Identity And Consent Management
 
+## Table of Contents
+
+- [v0.2.0-rc](#v020-rc)
```
Comment (hdamker) [inline CHANGELOG.md:5]:
```suggestion
- [v0.2.0-rc1](#v020-rc)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,51 @@
 # Changelog Identity And Consent Management
 
+## Table of Contents
+
+- [v0.2.0-rc1](#v020-rc)
```
Comment (rartych) [inline CHANGELOG.md:5]: The link should be like that:

```suggestion
- [v0.2.0-rc.1](#v020-rc1)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,51 @@
...
+- [v0.1.0 - Initial version](#v010---initial-version)
+
+# v0.2.0-rc.1
```
Comment (hdamker) [inline CHANGELOG.md:8]: this is correct ... "v0.2.0-rc.1"

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,51 @@
...
+
+**This is the release candidate version for "Identity And Consent Management" release 0.2.0**
+
+## Please note:
+
+* The working group agreed to proceed directly with a **Release Candidate** version, bypassing the **ALPHA** release, given the stability and closed scope of the 0.2.0 release.
+* The content of the release includes the "Identity And Consent Management" approved deliverables in **[documentation](https://github.com/camaraproject/IdentityAndConsentManagement/tree/release-0.2.0-rc/documentation)** folder.
```
Comment (hdamker) [inline CHANGELOG.md:15]:
```suggestion
* The content of the release includes the "Identity And Consent Management" approved deliverables in **[documentation](https://github.com/camaraproject/IdentityAndConsentManagement/tree/v0.2.0-rc.1/documentation)** folder.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,51 @@
...
+* The document [Authentication and Authorization Concept for Service APIs](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.1.0/documentation/CAMARA-AuthN-AuthZ-Concept.md) is part of the 0.1.0 release. **It is deprecated**. It will be removed after the 0.2.0 public release of "Identity and Consent Management".
+
+### Main Changes
+* Creation of the [CAMARA Security and Interoperability Profile](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.2.0-rc/documentation/CAMARA-Security-Interoperability.md) document.
```
Comment (hdamker) [inline CHANGELOG.md:19]:
```suggestion
* Creation of the [CAMARA Security and Interoperability Profile](https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-Security-Interoperability.md) document.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,51 @@
...
+* Creation of the [Identity and Consent Management Examples](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.2.0-rc/documentation/CAMARA-ICM-examples.md) document.
```
Comment (hdamker) [inline CHANGELOG.md:20]:
```suggestion
* Creation of the [Identity and Consent Management Examples](https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-ICM-examples.md) document.
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,51 @@
...
+* Aligment of the [CAMARA APIs access and user consent management](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.2.0-rc/documentation/CAMARA-API-access-and-user-consent.md) document with the latest decisions of the working group in the new profile.
```
Comment (hdamker) [inline CHANGELOG.md:21]:
```suggestion
* Aligment of the [CAMARA APIs access and user consent management](https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-API-access-and-user-consent.md) document with the latest decisions of the working group in the new profile.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,51 @@
...
+
+### Added
+* Added paragraph describing the handling on authorization flow selection during API product ordering in the [`CAMARA-API-access-and-user-consent.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.2.0-rc/documentation/CAMARA-API-access-and-user-consent.md) document by @Elisabeth-Ericsson in https://github.com/camaraproject/IdentityAndConsentManagement/pull/120
```
Comment (hdamker) [inline CHANGELOG.md:24]:
```suggestion
* Added paragraph describing the handling on authorization flow selection during API product ordering in the [`CAMARA-API-access-and-user-consent.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-API-access-and-user-consent.md) document by @Elisabeth-Ericsson in https://github.com/camaraproject/IdentityAndConsentManagement/pull/120
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,51 @@
...
+* Added the [`CAMARA-Security-Interoperability.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.2.0-rc/documentation/CAMARA-Security-Interoperability.md) profile document by @AxelNennker in https://github.com/camaraproject/IdentityAndConsentManagement/pull/121
```
Comment (hdamker) [inline CHANGELOG.md:25]:
```suggestion
* Added the [`CAMARA-Security-Interoperability.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-Security-Interoperability.md) profile document by @AxelNennker in https://github.com/camaraproject/IdentityAndConsentManagement/pull/121
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,51 @@
...
+* Added the [`CAMARA-ICM-examples.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.2.0-rc/documentation/CAMARA-ICM-examples.md) document by @AxelNennker in https://github.com/camaraproject/IdentityAndConsentManagement/pull/148
```
Comment (hdamker) [inline CHANGELOG.md:26]:
```suggestion
* Added the [`CAMARA-ICM-examples.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-ICM-examples.md) document by @AxelNennker in https://github.com/camaraproject/IdentityAndConsentManagement/pull/148
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,51 @@
...
+* Added 2-legged/3-legged access token definition to [`CAMARA-API-access-and-user-consent.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.2.0-rc/documentation/CAMARA-API-access-and-user-consent.md) document by @jpengar in https://github.com/camaraproject/IdentityAndConsentManagement/pull/162
```
Comment (hdamker) [inline CHANGELOG.md:27]:
```suggestion
* Added 2-legged/3-legged access token definition to [`CAMARA-API-access-and-user-consent.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-API-access-and-user-consent.md) document by @jpengar in https://github.com/camaraproject/IdentityAndConsentManagement/pull/162
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,51 @@
...
+  
+### Changed
+* Clarified resource server terminology by @Elisabeth-Ericsson in https://github.com/camaraproject/IdentityAndConsentManagement/pull/135
+* Updated the [`CAMARA-API-access-and-user-consent.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.2.0-rc/documentation/CAMARA-API-access-and-user-consent.md) document with the latest decisions of the working group in the new profile by @jpengar in https://github.com/camaraproject/IdentityAndConsentManagement/pull/155
```
Comment (hdamker) [inline CHANGELOG.md:31]:
```suggestion
* Updated the [`CAMARA-API-access-and-user-consent.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-API-access-and-user-consent.md) document with the latest decisions of the working group in the new profile by @jpengar in https://github.com/camaraproject/IdentityAndConsentManagement/pull/155
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,51 @@
...
+* Adapted the `info.description` template in [`CAMARA-API-access-and-user-consent.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.2.0-rc/documentation/CAMARA-API-access-and-user-consent.md) document to "CAMARA Security and Interoperability Profile" by @AxelNennker in https://github.com/camaraproject/IdentityAndConsentManagement/pull/168
```
Comment (hdamker) [inline CHANGELOG.md:32]:
```suggestion
* Adapted the `info.description` template in [`CAMARA-API-access-and-user-consent.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-API-access-and-user-consent.md) document to "CAMARA Security and Interoperability Profile" by @AxelNennker in https://github.com/camaraproject/IdentityAndConsentManagement/pull/168
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,51 @@
...
+
+### Fixed
+* Restored [`CAMARA-AuthN-AuthZ-Concept.md`]((https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.1.0/documentation/CAMARA-AuthN-AuthZ-Concept.md)) document with deprecation disclaimer by @hdamker in https://github.com/camaraproject/IdentityAndConsentManagement/pull/147
```
Comment (hdamker) [inline CHANGELOG.md:35]: I would link here to the version within the current release, as the v0.1.0 version does not include the deprecation disclaimer:

```suggestion
* Restored [`CAMARA-AuthN-AuthZ-Concept.md`]((https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-AuthN-AuthZ-Concept.md)) document with deprecation disclaimer by @hdamker in https://github.com/camaraproject/IdentityAndConsentManagement/pull/147
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,51 @@
...
+* Fixed Auth code flow error scenario when user refuses consent in [`CAMARA-API-access-and-user-consent.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/release-0.2.0-rc/documentation/CAMARA-API-access-and-user-consent.md) document by @jpengar in https://github.com/camaraproject/IdentityAndConsentManagement/pull/170.
```
Comment (hdamker) [inline CHANGELOG.md:36]:
```suggestion
* Fixed Auth code flow error scenario when user refuses consent in [`CAMARA-API-access-and-user-consent.md`](https://github.com/camaraproject/IdentityAndConsentManagement/blob/v0.2.0-rc.1/documentation/CAMARA-API-access-and-user-consent.md) document by @jpengar in https://github.com/camaraproject/IdentityAndConsentManagement/pull/170.
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,5 +1,51 @@
...
+
+### Removed
+* N/A
+
+## New Contributors
+* @Elisabeth-Ericsson made their first contribution in https://github.com/camaraproject/IdentityAndConsentManagement/pull/120
+* @AxelNennker made their first contribution in https://github.com/camaraproject/IdentityAndConsentManagement/pull/121
+* @hdamker made their first contribution in https://github.com/camaraproject/IdentityAndConsentManagement/pull/147
+
+**Full Changelog**: https://github.com/camaraproject/IdentityAndConsentManagement/compare/v0.1.0...v0.2.0-rc
```
Comment (hdamker) [inline CHANGELOG.md:46]:
```suggestion
**Full Changelog**: https://github.com/camaraproject/IdentityAndConsentManagement/compare/v0.1.0...v0.2.0-rc.1
```

**Review by hdamker — CHANGES_REQUESTED** [review]
The PR looks good for me. Just suggest the change of all links:
- the tag should be v0.2.0-rc.1 instead of v0.2.0-rc
- we dropped the creation of "release" branches (especially for pre-releases), instead the links should go directly to the tag within the blob/tree.

Fingers crossed that my suggested links are correct after the tag v0.2.0-rc.1 is created ...

**Review by diegogonmar — APPROVED** [review]
LGTM. Thanks for the adjusts

**Comment by hdamker** [comment]
@AxelNennker 
a) there are three "hidden" suggestions in my review to be committed to update the remaining links
b) I've updated the description of the PR slightly to align it with the versioning and release process guidelines

**Comment by tanjadegroot** [comment]
For the release process, please add @camaraproject/TSC_participants to the reviewers of this PR.

**Review by sebdewet — APPROVED** [review]
LGTM

**Review by hdamker — APPROVED** [review]
Thanks a lot @jpengar and @AxelNennker for the pull request. It looks good for me.

Within TSC today ([draft minutes](https://wiki.camaraproject.org/display/CAM/2024-07-04+TSC+Minutes)) we have decided to keep the release candidate PRs for Commonalities and ICM open until Wednesday, July 10th, for a final review including the TSC participants. So please wait with the merge until then.

**Review by bigludo7 — APPROVED** [review]

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Review by eric-murray — APPROVED** [review]

**Review by rartych — APPROVED** [review]
It can be merged now.

**Comment by hdamker** [comment]
@AxelNennker @jpengar Can one of you please merge this PR and create the pre-release in GitHub before today's AHC on 16:00? If you can't for some reason I offer to take it.

**Comment by hdamker** [comment]
@jpengar Thanks! Looks good to me. I would just recommend to update the README.md and mention there the release candidate explicitly, maybe even replacing the link to "latest public release" v0.1.0, as now the v0.2.0-rc.1 is more important.

**Comment by jpengar** [comment]
> @jpengar Thanks! Looks good to me. I would just recommend to update the README.md and mention there the release candidate explicitly, maybe even replacing the link to "latest public release" v0.1.0, as now the v0.2.0-rc.1 is more important.

@hdamker @AxelNennker @sebdewet https://github.com/camaraproject/IdentityAndConsentManagement/pull/179 please review.

**Pull request merged**